code/modules/clothing/glasses/glasses.dm

```diff
@@ -257,6 +257,9 @@
 		if(noir_mode)
 			if(color_view && user.client && !user.client.color)
 				animate(user.client, color = color_view, time = 10)
+	else
+		if(user.client && user.client.color)

```

Comment:
I thought that as well but I decided to drop it because I had my doubts, but I'm happy to put it back in (in the hopes it gets put to use some day, haha). Thanks for the input, sorry for the delay